src/index.js

```diff
@@ -149,6 +149,8 @@ class Dropzone extends React.Component {
       className,
       rejectStyle,
       style,
+      disablePreview, // eslint-disable-line no-unused-vars, prefer-const
+      disableClick, // eslint-disable-line no-unused-vars, prefer-const
```

Comment:
Since these variables aren't used here I'm wondering if it is better to use the last method with `delete` props from the cloned object here:

```
function MyDiv(props) {

  const divProps = Object.assign({}, props);
  delete divProps.layout;

  if (props.layout === 'horizontal') {
    return <div {...divProps} style={getHorizontalStyle()} />
  } else {
    return <div {...divProps} style={getVerticalStyle()} />
  }
}
```

See https://facebook.github.io/react/warnings/unknown-prop.html

What do you think? Mind updating the PR? Also, could you add a test that checks that these props aren't removed from the original `this.props`?